diff --git a/bbox.c b/bbox.c
index b7f2f7e..8d5078b 100644
--- a/bbox.c
+++ b/bbox.c
@@ -479,8 +479,8 @@
    real off = 72.0 / conv;
    real margin = 1.0 / conv;
    real vsize = 792.0 / conv;
-   real hadj = -72.0 * hoff / 4736286.72;
-   real vadj = 72.0 * voff / 4736286.72;
+   real hadj = -72.0 * hoff / 4736286.72 + 72.0;
+   real vadj = 72.0 * voff / 4736286.72 - 72.0;
 
    fseek(dvifile, bop, 0);
    bbdopage();
diff --git a/dopage.c b/dopage.c
index 991ca72..25e97ce 100644
--- a/dopage.c
+++ b/dopage.c
@@ -69,8 +69,8 @@
    H_BREAK = (30 * DPI / 400 ); /* 30 seems to have worked well at 400 dpi */
 #endif 
    w = x = y = z = dir = rdir = fdir = 0;
-   h = (integer) (DPI / conv * hoff / 4736286.72);
-   v = (integer) (DPI / conv * voff / 4736286.72);
+   h = (integer) (DPI / conv * hoff / 4736286.72 - DPI / conv);
+   v = (integer) (DPI / conv * voff / 4736286.72 - DPI / conv);
    hh = PixRound(h);
    vv = PixRound(v);
    curfnt = NULL;
diff --git a/dvips.c b/dvips.c
index 5c7f6d3..d8a0392 100644
--- a/dvips.c
+++ b/dvips.c
@@ -583,6 +583,7 @@
 #ifdef VMS
 main(void)
 #else
+char *myptr;
 int
 main(int argc, char **argv)
 #endif
@@ -879,7 +880,7 @@
                   p = argv[++i];
                handlepapersize(p, &hoff, &voff);
                break;
-case 'T' :
+case 'p' :
                if (*p == 0 && argv[i+1])
                   p = argv[++i];
                handlepapersize(p, &hpapersize, &vpapersize);
@@ -890,48 +891,14 @@
                }
                lastpsizwins = 0;
                break;
-case 'p' :
-#if defined(MSDOS) || defined(OS2) || defined(ATARIST)
-               /* check for emTeX job file (-pj=filename) */
-               if (*p == 'j') {
-                 p++;
-                 if (*p == '=' || *p == ':')
-                   p++;
-                 mfjobname = newstring(p);
-                 break;
-               }
-               /* must be page number instead */
-#endif
-               if (*p == 'p') {  /* a -pp specifier for a page list? */
-                  p++;
-                  if (*p == 0 && argv[i+1])
-                     p = argv[++i];
-                  if (ParsePages(p))
-                     error("! Bad page list specifier (-pp).");
-                  pagelist = 1;
-                  break;
-               }
+case 's' :
                if (*p == 0 && argv[i+1])
                   p = argv[++i];
-               if (*p == '=') {
-                  abspage = 1;
-                  p++;
-               }
-#ifdef SHORTINT
-               switch(sscanf(p, "%ld.%ld", &firstpage, &firstseq)) {
-#else        /* ~SHORTINT */
-               switch(sscanf(p, "%d.%d", &firstpage, &firstseq)) {
-#endif        /* ~SHORTINT */
-case 1:           firstseq = 0;
-case 2:           break;
-default:
-#ifdef KPATHSEA
-                  error(concat3 ("! Bad first page option (-p ", p, ")."));
-#else
-                  error("! Bad first page option (-p).");
-#endif
-               }
-               notfirst = 1;
+	       if (*p == '-' && *(p+1) == 0) break;
+               if (ParsePages(p))
+                  error("! Bad page list specifier (-s).");
+               pagelist = 1;
+	       abspage = 1;
                break;
 case 'l':
                if (*p == 0 && argv[i+1])
@@ -984,14 +951,18 @@
                 printf ("%s %s\n", banner, banner2);
                 exit (0);
                 break;
-case 'x' : case 'y' :
-               if (*p == 0 && argv[i+1])
-                  p = argv[++i];
-               if (sscanf(p, "%lg", &mag)==0 || mag < 1 ||
-                          mag > 1000000)
-                  error("! Bad magnification parameter (-x or -y).");
-               overridemag = (c == 'x' ? 1 : -1);
-               break;
+case 'x' :
+		if (*p == 0 && argv[i+1])
+		   p = argv[++i];
+                p[strlen(p)-2]=0;
+		hoff = strtol(p,&myptr,0);
+		break;
+case 'y' :
+		if (*p == 0 && argv[i+1])
+		   p = argv[++i];
+		p[strlen(p)-2]=0;
+		voff = strtol(p,&myptr,0);
+		break;
 case 'C' :
                if (*p == 0 && argv[i+1])
                   p = argv[++i];
@@ -1078,9 +1049,6 @@
                   }
 	       }
                break;
-case 's':
-               safetyenclose = (*p != '0');
-               break;
 case 'V':
                downloadpspk = (*p != '0');
                break;
